src/server/scripts/Commands/cs_guild.cpp

```diff
@@ -180,6 +181,65 @@ class guild_commandscript : public CommandScript
         uint8 newRank = uint8(atoi(rankStr));
         return targetGuild->ChangeMemberRank(targetGuid, newRank);
     }
+
+    static bool HandleGuildInfoCommand(ChatHandler* handler, char const* args)
+    {
+        Player* target;
+        uint32 guildId;
+        std::string guildName;
+        std::string guildMasterName;
+        Guild* guild;
+
+        if (!*args)
+        {
+            // Look for the guild of the selected player or ourselves
+            if (target = handler->getSelectedPlayerOrSelf())
+                guild = target->GetGuild();
+            else
+                // getSelectedPlayerOrSelf will return null if there is no session
+                // so target becomes nullptr if the command is ran through console
+                // without specifying args.
+                return false;
+        }
+        else if (guildId = atoi(args)) // Try searching by Id
+            guild = sGuildMgr->GetGuildById(guildId);
+        else
+        {
+            // Try to extract a guild name
+            char* tailStr = *args != '"' ? strtok(nullptr, "") : (char*)args;
```

Comment:
`strtok(nullptr)` expects a previous invocation of `strtok` with some pointer. You do not do this. Or am I missing it?
http://en.cppreference.com/w/cpp/string/byte/strtok

Reply:
Yes, it's better other way. I will commit again for a better version of this.